app/controllers/users/confirmations_controller.rb

```diff
@@ -0,0 +1,74 @@
+class Users::ConfirmationsController < Devise::ConfirmationsController
+  skip_before_action :authenticate_user!
+
+  # PUT /resource/confirmation
+  def update
+    with_unconfirmed_confirmable do
+      if @confirmable.has_no_password?
+        @confirmable.attempt_set_password(params[:user])
+        if @confirmable.valid? and passwords_match?
+          do_confirm
+        else
+          do_show
+          @confirmable.errors.clear #so that we wont render :new
```

Comment:
Is this comment a truth?

We use `@confirmable.errors.clear` as some kind of a hack to not render a `:new` view at all? 
Or is it "so that we won't render confirmable errors in the :new view" ?

Looks like the second case to me though I might not understand how this trick works 😉